phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKeyIT.java

```diff
@@ -754,6 +817,28 @@ private void assertRow(Connection conn, String tableName, String expectedPK, int
         assertFalse(rs.next());
     }
 
+    private void upsertRecord(Connection conn, String dml, String orgid, String userid, long ts, String status) throws SQLException {
+        try(PreparedStatement stmt = conn.prepareStatement(dml)) { // regular upsert
+            stmt.setString(1, orgid);
+            stmt.setString(2, userid);
+            stmt.setDate(3, new Date(ts));
+            stmt.setString(4, status); // status should change now
+            stmt.executeUpdate();
+            conn.commit();
+        }
+    }
+
+    private void assertNumRecords(int count, Connection conn, String dql, String... params)
+        throws Exception {
+        PreparedStatement stmt = conn.prepareStatement(dql);
+        int counter = 1;
+        for (String param : params) {
+            stmt.setString(counter++, param);
+        }
+        ResultSet rs = stmt.executeQuery();
+        assertTrue(rs.next());
+        assertEquals(count, rs.getInt(1));
```

Comment:
from the jira description looks like we want to make sure correct timestamps are used for the mutation but we are not validating that in the test case and just the row count. Is that sufficient @tkhurana ?

Reply:
@ankitjain64 I was able to repro the problem with just checking row count but I will add the asserts for timestamps also